test/distributed/org/apache/cassandra/distributed/test/JVMDTestTest.java

```diff
@@ -50,4 +56,30 @@ public void insertTimestampTest() throws IOException
             assertEquals(1000, (long) res[0][0]);
         }
     }
+
+    @Test
+    public void instanceLogs() throws IOException, TimeoutException
+    {
+        try (Cluster cluster = init(Cluster.build(2).withConfig(c -> c.with(Feature.values())).start()))
+        {
+            // debug logging is turned on so we will see debug logs
+            Assertions.assertThat(cluster.get(1).logs().grep("^DEBUG")).isNotEmpty();
+            // make sure an exception is thrown in the cluster
+            LogAction logs = cluster.get(2).logs();
+            long mark = logs.mark(); // get the current position so watching doesn't see any previous exceptions
+            cluster.get(2).runOnInstance(() -> {
+                // pretent that an uncaught exception was thrown
+                CassandraDaemon.uncaughtException(Thread.currentThread(), new RuntimeException("fail without fail"));
+            });
+            List<String> errors = logs.watchFor(mark, "^ERROR");
+            Assertions.assertThat(errors)
+                      // can't check for "fail without fail" since thats on the next line, and watchFor doesn't
+                      // stitch lines togher like grepForError does
```

Comment:
Nit: together?

Just to make sure I understand: we can check for `fail withouot fail`, we just can't do it with the same `watchFor`, right?

Reply:
Correct, this was trying to match the semantics of python dtest (aka ccm).  grepForErrors stitches logs together but watchFor didn't.

ccm watch for https://github.com/riptano/ccm/blob/master/ccmlib/node.py#L504-L512
ccm grep for errors https://github.com/riptano/ccm/blob/5c7b782bf072368480fb81fd1ef70a57b09b8c5a/ccmlib/node.py#L2098-L2118

watchFor matches a single line, which might be a single `    at class.method(42)`, or the start of a log line, or the exception message.